lib/email/sender.rb

```diff
@@ -367,12 +367,13 @@ def add_attachments(*posts, is_digest: false)
         post.uploads.each do |original_upload|
           optimized_1X = original_upload.optimized_images.first
 
-          # only attach images in digests
-          next if is_digest && !FileHelper.is_supported_image?(original_upload.original_filename)
-
-          if FileHelper.is_supported_image?(original_upload.original_filename) &&
-               !should_attach_image?(original_upload, optimized_1X)
-            next
+          if FileHelper.is_supported_image?(original_upload.original_filename)
+            next if !should_attach_image?(original_upload, optimized_1X)
+            # Don't attach images that aren't rendered in the e-mail.
+            next if is_digest && !@message.html_part.body.include?(original_upload.sha1)
```

Comment:
Simple, but works. There's a 0% chance the SHA1 would incidentally be in the e-mail body, so I don't think we need to be more sophisticated than this.